nifi-nar-bundles/nifi-mongodb-bundle/nifi-mongodb-processors/src/main/java/org/apache/nifi/processors/mongodb/GetMongo.java

```diff
@@ -121,34 +136,53 @@
             .addValidator(StandardValidators.POSITIVE_INTEGER_VALIDATOR)
             .build();
 
-    static final PropertyDescriptor BATCH_SIZE = new PropertyDescriptor.Builder()
-            .name("Batch Size")
-            .description("The number of elements returned from the server in one batch")
+    static final PropertyDescriptor FETCH_SIZE = new PropertyDescriptor.Builder()
+            .name("Fetch Size")
```

Comment:
Changing a property's name will cause it to be recognized as a "different" property. This will cause all existing flows containing GetMongo to become invalid (the "Batch Size" property will show up at the bottom containing the existing value, but the framework will claim the processor is invalid because Batch Size is not a supported property).
That's why we use displayName() for the user-friendly name, so we can change it at will. I realize you did not have that luxury here, but we still would have to keep the name("Batch Size") and add displayName("Fetch Size"). This will be confusing in the code (until we change it for real, perhaps in NiFi 2.0?) but can be accompanied by documentation.
Also I'm still a little leery of changing the existing property to "fetch" vs "batch", then using "batch" in a different context in the added property. Would like to get some input from others on this as well.

Reply:
I missed that it was `name` and not `displayName`.

Maybe what should be done here is to revert that change, and then have the commits happen either after reach flowfile (when grouped into big flowfiles) or after each batch as defined in that property for the 1:1 result/flowfile option.